libscript/src/script-builder.cpp

```diff
@@ -1394,36 +1382,26 @@ void MCScriptEndInvokeInModule(MCScriptModuleBuilderRef self)
     __end_instruction(self);
 }
 
-void MCScriptEmitFetchLocalInModule(MCScriptModuleBuilderRef self, uindex_t p_dst_reg, uindex_t p_local_index)
+void MCScriptEmitFetchInModule(MCScriptModuleBuilderRef self, uindex_t p_dst_reg, uindex_t p_index, uindex_t p_level)
 {
     if (self == nil || !self -> valid)
         return;
     
-    __emit_instruction(self, kMCScriptBytecodeOpFetchLocal, 2, p_dst_reg, p_local_index);
-}
-
-void MCScriptEmitStoreLocalInModule(MCScriptModuleBuilderRef self, uindex_t p_src_reg, uindex_t p_local_index)
-{
-    if (self == nil || !self -> valid)
-        return;
-    
-    __emit_instruction(self, kMCScriptBytecodeOpStoreLocal, 2, p_src_reg, p_local_index);
-}
-
-void MCScriptEmitFetchGlobalInModule(MCScriptModuleBuilderRef self, uindex_t p_dst_reg, uindex_t p_glob_index)
-{
-    if (self == nil || !self -> valid)
-        return;
-    
-    __emit_instruction(self, kMCScriptBytecodeOpFetchGlobal, 2, p_dst_reg, p_glob_index);
+    if (p_level == 0)
+        __emit_instruction(self, kMCScriptBytecodeOpFetch, 2, p_dst_reg, p_index);
```

Comment:
I'm not sure about this - they are basically arg counts so __emit_instruction knows how many parameters to encode for the bytecode op. I couldn't use a termination pattern really since both positive and negative integers are allowed (IIRC at least).